services/ui_backend_service/data/db/tables/run.py

```diff
@@ -135,57 +111,42 @@ def select_columns(self):
             WHEN end_attempt_ok IS NOT NULL
             THEN end_attempt_ok.ts_epoch
             WHEN {table_name}.last_heartbeat_ts IS NOT NULL
-                AND latest_failed_task IS NOT NULL
-                AND @(extract(epoch from now())-{table_name}.last_heartbeat_ts)>{heartbeat_threshold}
-            THEN {table_name}.last_heartbeat_ts*1000
-            WHEN {table_name}.last_heartbeat_ts IS NOT NULL
-                AND latest_failed_task IS NULL
-                AND @(extract(epoch from now())-{table_name}.last_heartbeat_ts)>{heartbeat_cutoff}
-            THEN {table_name}.last_heartbeat_ts*1000
-            ELSE NULL
+                AND @(extract(epoch from now())-{table_name}.last_heartbeat_ts)<={heartbeat_threshold}
```

Comment:
previous case had:
  - when you have a HB AND you have a latest failed task AND you have not had a HB recently (within threshold), then finished_at = last_hb
  - when you have a HB AND you have no last failed task that you know of AND you have not had a HB recentlyy (within CUTOFF -- not threshold -- cutoff is larger), then finished_at = last_hb
  - else no finished at

I changed that to remove the need for latest failed task and just check the last hb. If I have not had one in threshold time, I consider that I have a finished_at.